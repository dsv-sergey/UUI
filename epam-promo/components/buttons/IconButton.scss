@use  '../../assets/styles/index' as *;

.root {
    @include button-reset();
    cursor: pointer;
    fill: var(--icon-color);

    &:global(.-clickable) {
        &:hover {
            fill: var(--icon-color-hover);
        }

        &:active {
            fill: var(--icon-color-active);
        }
    }

    &:global(.uui-disabled) {
        fill: var(--icon-color-disabled);
        cursor: default;

        &:global(.-clickable) {
            &:hover {
                fill: var(--icon-color-disabled);
            }

            &:active {
                fill: var(--icon-color-disabled);
            }
        }
    }

    &:global(.uui-invalid) {
        fill: var(--icon-color-invalid);
    }
}
