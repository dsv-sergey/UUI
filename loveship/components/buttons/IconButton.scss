@use '../../assets/styles/scss/index'  as *;

.root {
    @include button-reset();
    overflow: visible;
    cursor: pointer;
    fill: var(--uui-color-700);

    &:global(.-clickable) {
        &:hover {
            fill: var(--uui-color-800);
        }

        &:active {
            fill: var(--uui-color-900);
        }
    }

    &:global(.uui-disabled) {
        fill: $night500;
        cursor: default;

        &:hover {
            fill: $night500;
        }
    }

    &:global(.uui-invalid) {
        fill: $fire;
    }
}
