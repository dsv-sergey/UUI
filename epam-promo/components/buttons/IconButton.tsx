import * as React from 'react';
import { ButtonBaseProps, ButtonBase } from '@epam/uui-components';
import { IconContainer } from '@epam/uui-components';
import * as css from './IconButton.scss';
import * as styles from '../../assets/styles/colorvars/buttons/iconButton-colorvars.scss';
import { allEpamPrimaryColors, EpamPrimaryColor } from '../types';

export type IconColor = EpamPrimaryColor | 'gray30' | 'gray50' | 'gray60';
export const allIconColors: IconColor[] = [...allEpamPrimaryColors, 'gray30', 'gray50', 'gray60'];

export interface IconButtonMods {
    color?: IconColor;
}

export interface IconButtonProps extends ButtonBaseProps, IconButtonMods { }

export class IconButton extends ButtonBase<IconButtonProps> {
    constructor(props: IconButtonProps) {
        super(props);
    }

    getClassName() {
        return [
            css.root,
            styles['icon-color-' + (this.props.color || 'gray60')],
        ];
    }

    getChildren() {
        return [
            <IconContainer key={ 'icon' } icon={ this.props.icon } />,
        ];
    }
}