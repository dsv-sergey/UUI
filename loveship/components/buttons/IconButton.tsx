import * as React from 'react';
import { ButtonBaseProps, ButtonBase } from '@epam/uui-components';
import { IconContainer } from '@epam/uui-components';
import * as types from '../types';
import * as css from './IconButton.scss';
import * as styles from '../../assets/styles/scss/loveship-color-vars.scss';

export interface IconButtonProps extends ButtonBaseProps, types.ColorMod { }

export class IconButton extends ButtonBase<IconButtonProps> {
    constructor(props: IconButtonProps) {
        super(props);
    }

    getClassName() {
        return [
            css.root,
            styles['color-' + (this.props.color || 'night600')],
        ];
    }

    getChildren() {
        return [
            <IconContainer key={ 'icon' } icon={ this.props.icon } />,
        ];
    }
}