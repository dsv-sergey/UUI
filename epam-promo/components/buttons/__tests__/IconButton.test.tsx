import React from 'react';
import { IconButton } from '../IconButton';
import renderer from 'react-test-renderer';
import * as calendarIcon from '../../../icons/calendar-18.svg';

describe('IconButton', () => {
    it('should be rendered correctly', () => {

        const tree = renderer
            .create(<IconButton />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer
            .create(<IconButton
                color='blue'
                onClick={ jest.fn }
                icon={ calendarIcon }
                isDisabled={ false }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});