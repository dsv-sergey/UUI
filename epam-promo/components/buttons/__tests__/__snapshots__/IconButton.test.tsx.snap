// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should be rendered correctly 1`] = `
<div
  className="root icon-color-gray60 uui-button-box uui-enabled"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <div
    className="container uui-icon uui-enabled"
  />
</div>
`;

exports[`IconButton should be rendered correctly with props 1`] = `
<div
  className="root icon-color-blue uui-button-box uui-enabled -clickable"
  href={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <div
    className="container uui-icon uui-enabled"
  >
    <svg
      className=""
      height={24}
      viewBox="0 0 12 24"
      width={12}
    >
      <use
        xlinkHref="#calendar-18.svg"
      />
    </svg>
  </div>
</div>
`;
